proto/server/keymanager/keymanager.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+package spire.server.keymanager;
+option go_package = "keymanager";
+
+import public "github.com/spiffe/spire/proto/common/plugin/plugin.proto";
+
+
+enum KeyAlgorithm {
+    UNSPECIFIED_KEY_ALGORITHM = 0;
+    ECDSA_P256 = 1;
+    ECDSA_P384 = 2;
+}
+
+enum HashAlgorithm {
+    UNSPECIFIED_HASH_ALGORITHM = 0;
+    // These entries (and their values) line up with a subset of the go
+    // crypto.Hash constants
+    SHA1 = 3;
```

Comment:
Should we support all of these? This one in particular? :)

Reply:
That's a great question and one I don't know that answer to :) I'd hate to have to rev key managers for each new hash we decide to use, but then again, i'm not sure how likely that will happen.

And you're right, the likeliness of needing a SHA1 signature is probably small :)